x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+    package:
+      name: epm/system
+      version: 1.7.0
+    inputs:
+      - type: metrics/system
+        streams:
+         - id?: {id}
+           enabled?: false # default true
+           metricset: cpu
+           period: 10s
+           dataset: system.cpu
+           metrics: ["percentages", "normalized_percentages"]
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+#################################################################################################
+### suggestion 1
+ - id: myendpoint-x1
+   title: Endpoint configuration
+   package:
+     name: endpoint
+     version: xxx
+   inputs:
+     - type: endpoint # Reserved key word
+       streams:
+         - type: malware
+           detect: true
+           prevent: false
+           notify_user: false
+           threshold: recommended
+           platform: windows
+
+        - type: eventing
+          api: true
+          clr: false
+          dll_and_driver_load: false
+          dns: true
+          file: false
+          platform: windows
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: mac
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: mac
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: linux
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: linux
+
+#################################################################################################
+### suggestion 2
+ - id: myendpoint-1
+   title: Endpoint configuration
+   package:
+     name: epm/endpoint # This establish the link with the package and will allow to link it to endpoint app.
+     version: xxx
+   inputs:
+     - type: endpoint # Reserved key word
```

Comment:
@kevinlog do you think it makes sense to have a separate "protections" section or similar? I imagine that there are a number of config settings that aren't really "inputs" or "streams". From my perspective, we don't need to over-generalize this Agent config in the sense that everything must fit as a stream. 

There's no reason why we need to make it all generic and the Agent oblivious to endgame - if that makes sense.

Reply:
Maybe `input` is stretching the concept for the endpoint, but they still fit out definition they generate data,  our requirements and reserved keywords are minimal everything else at the input level would be free form and passed directly to the endpoint. So they could have _protections_ or anything they want at that level.

```
- id: myendpoint-1
  title: Endpoint configuration
   package:
     name: epm/endpoint # This establish the link with the package and will allow to link it to endpoint app.
     version: xxx
   inputs:
     - type: endpoint # Reserved key word
```